server/Managers/Teams/TeamsManager.py

```diff
@@ -27,89 +30,86 @@ def __exit__(self, type, value, tb):
         # self.session.end_session()  <- mongodb mutex unlock
         pass
 
-    def get_id(self) -> str:
-        return self._id
-
-    def pass_data(self, data) -> None:
-        if data['members']:
-            self.add_members(list(set(data['members'])))
-        if data['eventID']:
-            self.team.join_event(data['eventID'])
-        if data['last_submission_timestamp']:
-            self.team.set_last_submission_timestamp(data['last_submission_timestamp'])
-
-    def create_team(self, owner: str) -> bool:
+    def create_team(self, teamdata: dict, owner: str) -> bool:
         if not self.found:
-            try:
-                self.team.set_owner(owner)
-                self.add_members([owner])
-                self.team.set_created_timestamp(datetime.now())
-                self.commit()
-                return True
-            except Exception:
-                return False
+            self.team.set_owner(owner)
+            self.team.set_members([owner] + (teamdata['members'].split()))
+            self.team.set_event_id(ObjectId(teamdata['event_id']))
+            self.team.set_created_timestamp(datetime.now())
+            self.commit()
+            return True
         else:
             return False
 
-    def update_team(self, team) -> bool:
+    def get_teams(self, email: str):
+        all_teams = list()
+        for team in self.db.find({'members': email}):
+            all_teams.append(team)
+        return all_teams
```

Comment:
Since the ids here from DB is ObjectID, you may want to stringify it since it will cause an error when return JSON object via API